src/store/ViewStore.ts

```diff
@@ -172,12 +175,14 @@ export default class ViewStore {
                 }
                 this.mainStore.state.setChartIsReady(true);
                 this.mainStore.state.setChartGranularity(granularity);
+                importFinishedPromise.resolve();
             };
             stx.importLayout(sortedItems, {
                 managePeriodicity: true,
                 preserveTicksAndCandleWidth: true,
                 cb: finishImportLayout,
             });
+            await importFinishedPromise;
```

Comment:
Using this promise to ensure that the code below is executed not before but after `finishImportLayout` is executed. The issue happened because the order of execution was random.